File: dart/bubble_sort.dart
List bubbleSort(List arr) {
  bool noSwaps;
  for (int i = 0; i < arr.length; i++) {
    noSwaps = true;
    for (int j = 0; j < arr.length - 1; j++) {
      if (arr[j] > arr[j + 1]) {
        int temp = arr[j + 1];
        arr[j + 1] = arr[j];
        arr[j] = temp;
        noSwaps = false;
      }
    }
    if (noSwaps) break;
  }

  return arr;
}

File: python/bubble_sort.py
def bubble_sort(arr):
  noSwaps = False
  i = 0
  while i < len(arr) and not noSwaps:
    j = 0
    noSwaps = True
    while j < len(arr) - 1:
      if arr[j] > arr[j + 1]:
        arr[j], arr[j + 1] = arr[j + 1], arr[j]
        noSwaps = False
      j += 1
    i += 1

  return arr